### ISSUE TITLE
[Bug]: PointAnnotation order cannot be managed (zIndex; symbolSortKey; key; id; sorting)

### ISSUE THREAD

**Issue by RyanTG** [rnmapbox/maps#3049]
### Mapbox Implementation

Mapbox

### Mapbox Version

default

### Platform

iOS

### `@rnmapbox/maps` version

10.0.13

### Standalone component to reproduce

```javascript
import React from 'react';
import { StyleSheet, View } from 'react-native';
import Mapbox from '@rnmapbox/maps';

const App = () => {
  return (
    <View style={styles.page}>
      <View style={styles.container}>
        <Mapbox.MapView 
          style={styles.map}
          scaleBarEnabled={false}
          pitchEnabled={false}
          rotateEnabled={false}
        >
          <Mapbox.PointAnnotation
            id={'10'}
            key={'10'}
            style={{zIndex: 10}}
            symbolSortKey={10}
            coordinate={[4.41, 5.87]}
          >
            <View
              style={{
                width: 30,
                height: 30,
                borderRadius: 15,
                borderWidth: 0,
                backgroundColor: 'yellow'
              }}
            />
          </Mapbox.PointAnnotation>
          <Mapbox.PointAnnotation
            id={'30'}
            key={'30'}
            style={{zIndex: 30}}
            symbolSortKey={30}
            coordinate={[4.51, 5.77]}
          >
            <View
              style={{
                width: 30,
                height: 30,
                borderRadius: 15,
                borderWidth: 0,
                backgroundColor: 'green'
              }}
            />
          </Mapbox.PointAnnotation>
          <Mapbox.PointAnnotation
            id={'20'}
            key={'20'}
            style={{zIndex: 20}}
            symbolSortKey={20}
            coordinate={[4.61, 5.87]}
          >
            <View
              style={{
                width: 30,
                height: 30,
                borderRadius: 15,
                borderWidth: 0,
                backgroundColor: 'red'
              }}
            />
          </Mapbox.PointAnnotation>
        </Mapbox.MapView>
      </View>
    </View>
  );
}

export default App;

const styles = StyleSheet.create({
  page: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  container: {
    height: '100%',
    width: '100%',
  },
  map: {
    flex: 1
  }
});
```

### Observed behavior and steps to reproduce

Example contains 3 PointAnnotation markers situated near one another. Zoom in a little and you can see how they overlap each other.

The observed behavior is that the order of the markers (how they overlap each others) is based on an unknown methodology, and methods to manage the order are not working.

### Expected behavior

I would expect `symbolSortKey` or `zIndex` or even the sorting of the PointAnnotations to affect the order.

### Notes / preliminary analysis

Methods include:
- Sorting the PointAnnotation components (as in, actually moving one before another in the code)
- Applying an `id` that denotes order
- Applying a `key` that denotes order
- Applying a `zIndex` to the style that denotes order
- Applying a `symbolSortKey` (which is a prop that appears to be available in both the Android and iOS code) that denotes order.

Other observed behavior is that when `pitchEnabled` or `rotateEnabled` are `true`, playing with those will end up changing the order (markers will suddenly hop on top of another).

At times I _thought_ I was observing the order being based on markers painted from lower right to upper left. But this doesn't appear to be consistent!

In summary, there seems to be no clear way to manage the order of the markers.

**Comment by mfazekas** [comment]
@RyanTG `symbolSortKey` and `symbolZOrder` applies for symbol layer. `PointAnnotation` doesn't have such properties see https://github.com/rnmapbox/maps/blob/main/docs/PointAnnotation.md

**Comment by RyanTG** [comment]
I suspected as much. But then when I saw

`symbolSortKey` on 

https://docs.mapbox.com/android/maps/api/10.16.0/mapbox-maps-android/com.mapbox.maps.plugin.annotation.generated/-point-annotation/
and
https://docs.mapbox.com/ios/maps/api/10.2.0/Structs/PointAnnotation.html

I thought perhaps there was a chance it was available.
